packages/block-library/src/navigation-link/edit.js

```diff
@@ -308,12 +308,14 @@ export default compose( [
 	withSelect( ( select, ownProps ) => {
 		const {
 			getBlockAttributes,
-			getBlockParents,
 			getClientIdsOfDescendants,
 			hasSelectedInnerBlock,
+			getBlockParentsByBlockName,
 		} = select( 'core/block-editor' );
 		const { clientId } = ownProps;
-		const rootBlock = getBlockParents( clientId )[ 0 ];
+		const rootBlock = head(
+			getBlockParentsByBlockName( clientId, 'core/navigation' )
```

Comment:
If I'm not mistaken, the`core/navigation` block is always the direct parent  to `navigaton-link` which means it's the last item in `getBlockParents(clientId)`

can't we solve this by just doing `last(getBlockParents( clientId ))` instead of introducing a new API (selector)?

Reply:
The issue happens when the Navigation gets nested items/submenus. Picking up just the first/last item is not enough.
![image](https://user-images.githubusercontent.com/77539/73856165-e4374180-47e9-11ea-94de-c4a1a479135b.png)